src/sqlfluff/core/rules/functional/segment_predicates.py

```diff
@@ -97,6 +97,15 @@ def _(segment: BaseSegment) -> bool:
     return _
 
 
+def is_templated() -> Callable[[BaseSegment], bool]:
+    """Returns a function that checks if segment is templated."""
+
+    def _(segment: BaseSegment) -> bool:
+        return segment.is_templated
```

Comment:
I don’t understand the need for this function, when it just seems to call the other function. Can’t that be called directly without the need of this function?

Reply:
It can. These functions exist to make it easy to work with the functional API. With this function, you can write:
```
            if not anchor_list.any(sp.not_(sp.is_templated())):
                continue
```

Without it, you'd have to write:
```
            if not anchor_list.any(lambda seg: not seg.is_templated):
                continue
```

If you make people use `lambda`, in my opinion, you've failed. 😄

Reply:
Note that above, we're not calling the function directly. We're calling a function that calls the function. That's where convenience and consistency of notation can be lost if you don't provide nice wrappers.